lib/segment/src/segment/segment_ops.rs

```diff
@@ -734,8 +734,10 @@ fn restore_snapshot_in_place(snapshot_path: &Path) -> OperationResult<()> {
 
 fn unpack_snapshot(segment_path: &Path) -> OperationResult<()> {
     let db_backup_path = segment_path.join(DB_BACKUP_PATH);
-    crate::rocksdb_backup::restore(&db_backup_path, segment_path)?;
-    std::fs::remove_dir_all(&db_backup_path)?;
+    if db_backup_path.is_dir() {
+        crate::rocksdb_backup::restore(&db_backup_path, segment_path)?;
+        std::fs::remove_dir_all(&db_backup_path)?;
+    }
```

Comment:
Why do we need this branch now?

Reply:
Partial snapshot might not contain RocksDB backup. E.g., if there was no changes to RocksDB.

Reply:
Ah, of course!

I thought it was either an extracted directory or a tar file. That I was confused by. But it is about something totally different.